src/api/middleware/files.js

```diff
@@ -99,3 +99,90 @@ export const cloudSave = async (req, res, next) => {
     return next();
 
 };
+
+export const parseArrayOfFiles = (field_name, max_count) => (req, res, next) => {
+    const upload = multerConfig.array(field_name, max_count);
+    upload(req, res, (error) => {
+        if (error || !req.files) {
+            let message = "required";
+            let param = field_name;
+            if (error) {
+                message = error instanceof MulterError ?
+                    parseError(error.message) : error.message;
+
+                if (error.code === "LIMIT_FILE_COUNT")
+                    message = ValidationReasons.ARRAY_SIZE(0, max_count);
+
+                param = error.field ? error.field : param;
+            }
+            return next(new APIError(
+                HTTPStatus.UNPROCESSABLE_ENTITY,
+                ErrorTypes.VALIDATION_ERROR,
+                [{
+                    location: "body",
+                    param,
+                    msg: message
+                }]
+
+            ));
+        } else {
+            return next();
+        }
+    });
+};
+
+export const localSaveArray = async (req, res, next) => {
+    for (const file of req.files) {
+        const buffer = file.buffer;
+        const extension = file.mimetype.substr(file.mimetype.indexOf("/") + 1);
+        const filename = `${req.user.company}.${extension}`;
```

Comment:
This line will make it so every file this company uploads has the same name. Maybe add a nonce or a sequence number to make the name unique.